addon/components/frost-toggle.js

```diff
@@ -78,42 +113,40 @@ export default Component.extend(PropTypeMixin, FrostEventsProxy, {
     return e
   },
 
+  // FIXME: jsdoc
   _preferBoolean (value) {
     if (value === 'true') return true
     if (value === 'false') return false
 
     return value
   },
 
+  // FIXME: jsdoc
   _setupAssertion () {
     assert(`Same value has been assigned to both ${this.toString()}.trueValue and ${this.toString()}.falseValue`,
       (typeOf(get(this, 'trueValue')) === 'undefined' && typeOf(get(this, 'falseValue')) === 'undefined') ||
       get(this, 'trueValue') !== get(this, 'falseValue'))
   },
 
-  // == Computed Properties =====================================================
-  @computed('trueValue', '_trueLabel')
-  _trueValue (trueValue, _trueLabel) {
-    return trueValue || _trueLabel
-  },
+  // == DOM Events ============================================================
 
-  @computed('falseValue', '_falseLabel')
-  _falseValue (falseValue, _falseLabel) {
-    return falseValue || _falseLabel
-  },
+  // == Lifecycle Hooks =======================================================
 
-  @computed('value')
-  _isToggled (value) {
-    return this._preferBoolean(value) === get(this, '_trueValue')
+  /* Ember.Component method */
+  init () {
+    this._super(...arguments)
+    this._setupAssertion()
   },
 
-  // == Actions ================================================================
+  // == Actions ===============================================================
+
   actions: {
     /* eslint-disable complexity */
+    // FIXME: eslint
     _onClick (event) {
       if (get(this, 'disabled')) return
 
-      if (!isSimpleClick(event)) {
+      if (!ViewUtils.isSimpleClick(event)) {
```

Comment:
Again, nested deconstructing makes my eyes bleed. But I can deconstruct it if you want. I don't believe we have any kind of rule that says everything must be deconstructed to the lowest possible level. The reason for deconstructing `Ember` was b/c they're soon gonna be doing named exports (or at least, that was my understanding). There is a downside when deconstructing as well, in that it is much more difficult, actually, normally impossible to stub the method in a test if you use it deconstructed.

If you want to stub this in a test, you can get a handle on `ViewUtils` in the test code and do something like
```js
sandbox.stub(ViewUtils, 'isSimpleClick').returns(true)
```
Whereas, if it's deconstructed at the top of the module, replacing the `isSimpleClick` on `VIewUtils` from the test module won't do you any good b/c the module under test already has a handle on the non-stubbed version and will call that instead.